### ISSUE TITLE
Model rendering fails

### ISSUE THREAD

**Issue by LeeviLindgren** [pyro-ppl/numpyro#1566]
Hi! I am not able to reproduce the example in [docs](https://num.pyro.ai/en/stable/tutorials/model_rendering.html). Running 

```
import numpyro.distributions as dist
import numpyro
import jax.numpy as jnp

assert numpyro.__version__.startswith("0.11.0")

def model(data):
    m = numpyro.sample("m", dist.Normal(0, 1))
    sd = numpyro.sample("sd", dist.LogNormal(m, 1))
    with numpyro.plate("N", len(data)):
        numpyro.sample("obs", dist.Normal(m, sd), obs=data)

data = jnp.ones(10)
numpyro.render_model(model, model_args=(data,))
```
results in 
```
NameError: unbound axis name: _provenance. The following axis names (e.g. defined by pmap) are available to collective operations: []
```

Environment:
```
> python --version
Python 3.10.3
> pip list
Package          Version
---------------- -------
graphviz         0.20.1
jax              0.4.7
jaxlib           0.4.7
ml-dtypes        0.0.4
multipledispatch 0.6.0
numpy            1.24.2
numpyro          0.11.0
opt-einsum       3.3.0
pip              22.0.4
scipy            1.10.1
setuptools       58.1.0
six              1.16.0
tqdm             4.65.0
```

Full error message:
```
NameError                                 Traceback (most recent call last)
Input In [1], in <cell line: 14>()
     11         numpyro.sample("obs", dist.Normal(m, sd), obs=data)
     13 data = jnp.ones(10)
---> 14 numpyro.render_model(model, model_args=(data,))

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/infer/inspect.py:631, in render_model(model, model_args, model_kwargs, filename, render_distributions, render_params)
    608 def render_model(
    609     model,
    610     model_args=None,
   (...)
    614     render_params=False,
    615 ):
    616     """
    617     Wrap all functions needed to automatically render a model.
    618 
   (...)
    629     :param bool render_params: Whether to show params in the plot.
    630     """
--> 631     relations = get_model_relations(
    632         model,
    633         model_args=model_args,
    634         model_kwargs=model_kwargs,
    635     )
    636     graph_spec = generate_graph_specification(relations, render_params=render_params)
    637     graph = render_graph(graph_spec, render_distributions=render_distributions)

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/infer/inspect.py:403, in get_model_relations(model, model_args, model_kwargs)
    393 params = {
    394     name: jax.tree_util.tree_map(
    395         lambda x: ProvenanceArray(x, frozenset({name})), site["value"]
   (...)
    398     if site["type"] == "param"
    399 }
    401 sample_and_params = {**samples, **params}
    402 sample_params_deps = get_provenance(
--> 403     eval_provenance(get_log_probs, sample_and_params)
    404 )
    406 sample_sample = {}
    407 sample_param = {}

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/ops/provenance.py:109, in eval_provenance(fun, *args, **kwargs)
    107 with jax.core.new_main(_ProvenanceJaxprTrace, dynamic=True) as main:
    108     main.jaxpr_stack = ()
--> 109     out = partial_eval.trace_to_subjaxpr_dynamic(fun, main, avals)[1]
    111 # unflatten the output and get its provenance
    112 out = [jax.ShapeDtypeStruct(x.shape, x.dtype, x.named_shape) for x in out]

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/interpreters/partial_eval.py:2066, in trace_to_subjaxpr_dynamic(fun, main, in_avals, keep_inputs, debug_info)
   2064 in_tracers = _input_type_to_tracers(trace.new_arg, in_avals)
   2065 in_tracers_ = [t for t, keep in zip(in_tracers, keep_inputs) if keep]
-> 2066 ans = fun.call_wrapped(*in_tracers_)
   2067 out_tracers = map(trace.full_raise, ans)
   2068 jaxpr, consts = frame.to_jaxpr(out_tracers)

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/_src/linear_util.py:166, in WrappedFun.call_wrapped(self, *args, **kwargs)
    163 gen = gen_static_args = out_store = None
    165 try:
--> 166   ans = self.f(*args, **dict(self.params, **kwargs))
    167 except:
    168   # Some transformations yield from inside context managers, so we have to
    169   # interrupt them before reraising the exception. Otherwise they will only
    170   # get garbage-collected at some later time, running their cleanup tasks
    171   # only after this exception is handled, which can corrupt the global
    172   # state.
    173   while stack:

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/_src/linear_util.py:166, in WrappedFun.call_wrapped(self, *args, **kwargs)
    163 gen = gen_static_args = out_store = None
    165 try:
--> 166   ans = self.f(*args, **dict(self.params, **kwargs))
    167 except:
    168   # Some transformations yield from inside context managers, so we have to
    169   # interrupt them before reraising the exception. Otherwise they will only
    170   # get garbage-collected at some later time, running their cleanup tasks
    171   # only after this exception is handled, which can corrupt the global
    172   # state.
    173   while stack:

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/infer/inspect.py:381, in get_model_relations.<locals>.get_log_probs(sample)
    379 for name, site in tr.items():
    380     if site["type"] == "sample":
--> 381         provenance_arrays[name] = site["fn"].log_prob(site["value"])
    382     elif site["type"] == "deterministic":
    383         provenance_arrays[name] = site["args"][0]

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/distributions/util.py:668, in validate_sample.<locals>.wrapper(self, *args, **kwargs)
    667 def wrapper(self, *args, **kwargs):
--> 668     log_prob = log_prob_fn(self, *args, **kwargs)
    669     if self._validate_args:
    670         value = kwargs["value"] if "value" in kwargs else args[0]

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/numpyro/distributions/continuous.py:2017, in Normal.log_prob(self, value)
   2014 @validate_sample
   2015 def log_prob(self, value):
   2016     normalize_term = jnp.log(jnp.sqrt(2 * jnp.pi) * self.scale)
-> 2017     value_scaled = (value - self.loc) / self.scale
   2018     return -0.5 * value_scaled**2 - normalize_term

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/_src/core.py:646, in Tracer.__sub__(self, other)
--> 646 def __sub__(self, other): return self.aval._sub(self, other)

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/_src/numpy/array_methods.py:251, in _defer_to_unrecognized_arg.<locals>.deferring_binary_op(self, other)
    249 args = (other, self) if swap else (self, other)
    250 if isinstance(other, _accepted_binop_types):
--> 251   return binary_op(*args)
    252 if isinstance(other, _rejected_binop_types):
    253   raise TypeError(f"unsupported operand type(s) for {opchar}: "
    254                   f"{type(args[0]).__name__!r} and {type(args[1]).__name__!r}")

    [... skipping hidden 12 frame]

File ~/virtualenvs/numpyro_test/lib/python3.10/site-packages/jax/_src/core.py:2474, in axis_frame(axis_name)
   2471     return frame
   2472 named_axes = [frame.name for frame in reversed(frames)
   2473               if not isinstance(frame.name, _TempAxisName)]
-> 2474 raise NameError(
   2475     f'unbound axis name: {axis_name}. The following axis names (e.g. defined '
   2476     f'by pmap) are available to collective operations: {named_axes}')

NameError: unbound axis name: _provenance. The following axis names (e.g. defined by pmap) are available to collective operations: []
```

**Comment by perone** [comment]
I'm getting the same all the time as well, even with very simple models.

**Comment by fehiepsi** [comment]
Hi @LeeviLindgren @perone this is fixed upstream. See #1555. I'll try to draft a new release in a few days.

**Comment by daeh** [comment]
Looking forward to this release. Thanks for the fix, @fehiepsi !

**Comment by fehiepsi** [comment]
Sorry for the delay. New release will be available after #1593 merged.
